vim-patch:8.1.{716,1139},8.2.{112,120,123}

janlazo opened the pull request:
8.1.1434 caused the failures so I'm leaving it for later.

janlazo committed:
vim-patch:8.2.0112: illegal memory access when using 'cindent'

Problem:    Illegal memory access when using 'cindent'.
Solution:   Check for NUL byte. (Dominique Pelle, closes vim/vim#5470)
https://github.com/vim/vim/commit/02ad46394e8f887b60fda994f8a5da2ac1937b23

justinmk commented:
travis/sourcehut:

```
From test_cindent.vim:
Found errors in Test_cindent_30():
function RunTheTest[37]..Test_cindent_30 line 32: Expected ['void', 'foo()', '{', 'if (a)', '{', '} else', '\tasdf;', '}', ''] but got ['void', 'foo()', '{', '\tif (a)', '\t{', '} else', '\tasdf;', '}', '']
Found errors in Test_cindent_34():
function RunTheTest[37]..Test_cindent_34 line 36: Expected ['', 'void', 'func(int a', '#if defined(FOO)', ' , int b', ' , int c', '#endif', ')', '{', '}', ''] but got ['', '\tvoid', 'func(int a', '#if defined(FOO)', '\t , int b', '\t , int c', '#endif', '\t)', '{', '}', '']
/home/build/neovim/build/bin/nvim -u unix.vim -U NONE -i NONE --noplugin --headless -u NONE -S summarize.vim messages
```

janlazo committed:
vim-patch:8.2.0120: virtcol() does not check arguments to be valid

Problem:    virtcol() does not check arguments to be valid, which may lead to
            a crash.
Solution:   Check the column to be valid.  Do not decrement MAXCOL.
            (closes vim/vim#5480)
https://github.com/vim/vim/commit/b3d33d8570bc49a7f90990572d7f9630a1bfae02

janlazo committed:
clang/'Logic error': zero-init struct

janlazo committed:
spell: spell_soundfold_sal() is dead code

janlazo committed:
vim-patch:8.1.0716: get warning message when 'completefunc' returns nothing

Problem:    Get warning message when 'completefunc' returns nothing.
Solution:   Allow for returning v:none to suppress the warning message.
            (Yasuhiro Matsumoto, closes vim/vim#3789)
https://github.com/vim/vim/commit/cee9bc2e3dc5c16a9d2a8d0e23aa0d5fdefa3a4a

janlazo committed:
vim-patch:8.1.1139: no test for what is fixed in patch 8.1.0716

Problem:    No test for what is fixed in patch 8.1.0716.
Solution:   Add a test. (Yasuhiro Matsumoto, closes vim/vim#3797)
https://github.com/vim/vim/commit/9845f36aa6ba28e0aa388bb635d4bb8ab56f1a47

janlazo committed:
vim-patch:8.2.0123: complete_info() does not work when CompleteDone is triggered

Problem:    complete_info() does not work when CompleteDone is triggered.
Solution:   Trigger CompleteDone before clearing the info.
https://github.com/vim/vim/commit/17e04781f26c24769e202351c194ee252927eee1
